jupyterhub/templates/hub/regsecret.yaml

```diff
@@ -0,0 +1,10 @@
+{{- if .Values.singleuser.image.createPullSecret }}
+kind: Secret
+apiVersion: v1
+metadata:
+  name: {{ .Values.singleuser.image.pullSecret.name }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {{ .Values.singleuser.image.pullSecret.dockerConfigJson }}
```

Comment:
I read in [the link](https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod) referenced by the [kubespawner documentation](https://github.com/jupyterhub/kubespawner/blob/master/kubespawner/spawner.py#L417-L431) that we are supposed to base64 encode the json...

Good thing, Helm will allows to do this quite easily I think. Oh so I googled for helm base64 and [found this](https://github.com/helm/helm/blob/master/docs/charts_tips_and_tricks.md#creating-image-pull-secrets) very relevant documentation!

From the looks of it, perhaps a config like this would make sense?

```yaml
singleuser:
  image:
    name: someimagename
    tag: sometag
    credentials:
      registry: quay.io
      username: someone
      password: sillyness
```

Reply:
so my thought here was someone with "pre" base64 encode this because it is a json blob and that could be a mess in the chart.yml. 

If this is against helm best practices i will gladly change it. 

I am new helm, we wrote our own kubernetes manifest/deploy at ezCater and don't leverage helm.

Reply:
I figure it would be quite easy to grasp for other users of the chart if we set it up like above. It would also be relatively easy to communicate and write the documentation what the users of the chart should do to set an image pull secret.

Reply:
I figure reducing the effort of the chart user is good enough as motivation to take care of the base64 encoding for the chart users, but I'm afraid that some chart users might think that the credentials are encrypted just because they were base64 encoded even though they are not as base64 is easy to decode. If we make the user write out the credentials in plain text the chart user can't be mislead.

Reply:
all of this makes sense will fix in my next commit.